sync-core/src/main/java/com/cloudant/sync/query/IndexManager.java

```diff
@@ -94,18 +93,32 @@ public IndexManager(Datastore datastore) {
         queue = Executors.newSingleThreadExecutor();
 
         final String filename = datastore.extensionDataFolder(EXTENSION_NAME) + File.separator
-                                                                        + "indexes.sqlite";
+                                                                              + "indexes.sqlite";
         SQLDatabase sqlDatabase = null;
         try {
             sqlDatabase = queue.submit(new Callable<SQLDatabase>() {
                 @Override
                 public SQLDatabase call() throws Exception {
                     SQLDatabase db = SQLDatabaseFactory.openSqlDatabase(filename);
-                    String[] schemaIndex = { "CREATE TABLE " + INDEX_METADATA_TABLE_NAME + " ( "
-                            + "        index_name TEXT NOT NULL, "
-                            + "        index_type TEXT NOT NULL, "
-                            + "        field_name TEXT NOT NULL, "
-                            + "        last_sequence INTEGER NOT NULL);" };
+                    String[] schemaIndex;
+                    if (db.getVersion() == 1) {
+                        // Update previous version of query metadata table to
+                        // support full text search.
+                        schemaIndex = new String[] {
+                                "ALTER TABLE " + INDEX_METADATA_TABLE_NAME + " "
+                                               + "ADD COLUMN index_settings TEXT NULL;"
+                        };
+                    } else {
+                        schemaIndex = new String[] {
+                                "CREATE TABLE " + INDEX_METADATA_TABLE_NAME + " ( "
+                                                + "index_name TEXT NOT NULL, "
+                                                + "index_type TEXT NOT NULL, "
+                                                + "index_settings TEXT NULL, "
+                                                + "field_name TEXT NOT NULL, "
+                                                + "last_sequence INTEGER NOT NULL);"
+                        };
+                    }
+
```

Comment:
Can we do the migrations here in the style of the datastore, running each migration after another?